lib/hawkular/prometheus/prometheus_api.rb

```diff
@@ -0,0 +1,50 @@
+require 'hawkular/base_client'
+require 'hawkular/inventory/entities'
+require 'ostruct'
+
+module Hawkular::Prometheus
+  class AlerterClient < Hawkular::BaseClient
+    def initialize(entrypoint, credentials = {}, options = {})
+      entrypoint = normalize_entrypoint_url entrypoint, 'hawkular/alerter'
+      @entrypoint = entrypoint
+
+      super(entrypoint, credentials, options)
+    end
+
+    def prometheus_entrypoint
+      rest_client('/prometheus/endpoint').get
+    end
+  end
+
+  # Interface to talk with the Prometheus server used for Middleware Manager
+  # @param entrypoint [String] base url of Hawkular Services
+  class Client < Hawkular::BaseClient
+    def initialize(entrypoint, credentials = {}, options = {})
+      prometheus_entrypoint = AlerterClient.new(entrypoint, credentials, options).prometheus_entrypoint
+      entrypoint = normalize_entrypoint_url prometheus_entrypoint, 'api/v1'
+      @entrypoint = entrypoint
```

Comment:
Use parens on methods that are included in an expression, on a first glance it was really hard to understand this because it reads like one expression instead of a method with two parameters.

Reply:
I guess I am just following the same pattern used in other clients.
Probably your comment is right, but I would try to be consistent in the meaning that potential refactoring in whole client could go in a separate PR.

Reply:
Disagree, I think we should get this one correct and refactor the other clients on a separate PR.